sycl/source/detail/sampler_impl.cpp

```diff
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <CL/sycl/context.hpp>
+#include <CL/sycl/detail/context_impl.hpp>
```

Comment:
only context class is used. Can we have farward declaration here?

Reply:
@vladimirlaz see https://github.com/intel/llvm/pull/873/files/55ec3d0c56e7555c720f9bdf190ec160739a584c#diff-ff508afff1be2b73d2e5cece4232ccc5R60

```
getSyclObjImpl(Context)->getHandleRef()
```
is an instance of `context_impl`, so we do need it.